AspNetCore3ChildlifetimeScope/Program.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Threading.Tasks;
+using AspNetCore3ChildlifetimeScope.Dependencies;
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+
+namespace AspNetCore3ChildlifetimeScope
+{
+    public static class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            var containerBuilder = new ContainerBuilder();
+
+            containerBuilder.RegisterType<SingletonRootDependency>()
+                .As<ISingletonRootDependency>()
+                .AsSelf()
+                .SingleInstance();
+
+            var container = containerBuilder.Build();
+
+            var scope = container.BeginLifetimeScope("RootChildScope", builder => builder.RegisterType<TransientRootChildScopeDependency>().As<ITransientRootChildScopeDependency>().AsSelf().InstancePerDependency());
```

Comment:
I think having this shared child scope here is both interesting and confusing at the same time. Like, it shows you can use a scope as the "root" for an app... but the apps are still both using this one shared root, which means it's confusing if I was trying to illustrate that there can be shared components but _also_ app-specifiic components.

(I do see the app-specific stuff getting registered, but why this is here isn't clear in the context of the rest of the example.)

I'd say _either_:

- Remove the `RootChildScope` to make the example simpler, _OR_
- Create two separate root child scopes, each with their own transient, and show each app can use a different child.